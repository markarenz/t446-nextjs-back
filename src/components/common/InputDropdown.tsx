import { SelectOption } from '../../types/types';
import styles from '../../styles/modules/InputDropdown.module.scss';

type Props = {
  name: string;
  value: string;
  onChange: Function;
  options: SelectOption[];
};
const InputDropdown: React.FC<Props> = ({ name, value, onChange, options }) => {
  return (
    <select
      name={name}
      value={value}
      onChange={(e) => onChange(e)}
      className={styles.inputDropdown}
    >
      {options.map((o) => (
        <option key={o.label} value={o.value}>
          {o.label}
        </option>
      ))}
    </select>
  );
};

export default InputDropdown;
