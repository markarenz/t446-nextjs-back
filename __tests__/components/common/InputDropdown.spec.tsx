import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import InputDropdown from '../../../src/components/common/InputDropdown';

const defaultProps = {
  name: 'fieldName',
  testId: 'test-dropdown',
  value: 'val1',
  onChange: jest.fn(),
  options: [
    { label: 'option 1', value: 'val1' },
    { label: 'option 2', value: 'val2' }
  ]
};
describe('InputDropdown', () => {
  it('renders default dropdown', async () => {
    render(<InputDropdown {...defaultProps} />);
    const element = await screen.findByTestId('test-dropdown');
    expect(element).toBeInTheDocument();
  });

  it('handles value change', async () => {
    render(<InputDropdown {...defaultProps} />);
    const element = await screen.findByTestId('test-dropdown');
    fireEvent.change(element, { target: { value: 'val2' } });
    expect(defaultProps.onChange).toHaveBeenCalled();
  });
});
