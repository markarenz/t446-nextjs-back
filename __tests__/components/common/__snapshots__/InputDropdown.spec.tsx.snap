// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDropdown renders default dropdown 1`] = `
<select
  class="inputDropdown"
  data-testid="test-dropdown"
  name="fieldName"
>
  <option
    value="val1"
  >
    option 1
  </option>
  <option
    value="val2"
  >
    option 2
  </option>
</select>
`;
